components/builder-scheduler/src/data_store.rs

```diff
@@ -262,6 +265,70 @@ impl DataStore {
+                                   owner_id bigint,
+                                   job_state text,
+                                   project_name text,
+                                   net_error_code int,
+                                   net_error_msg text,
+                                   build_started_at timestamptz,
+                                   build_finished_at timestamptz,
+                                   created_at timestamptz DEFAULT now()
+                            )"#,
+        )?;
+
+        // Insert a new job status
+        migrator.migrate("scheduler",
+                            r#"CREATE OR REPLACE FUNCTION insert_job_status_v1 (
+                               job_id bigint,
+                               owner_id bigint,
+                               job_state text,
+                               project_name text,
+                               net_error_code int,
+                               net_error_msg text,
+                               build_started_at timestamptz,
+                               build_finished_at timestamptz
+                           ) RETURNS SETOF job_status AS $$
+                                   BEGIN
+                                       RETURN QUERY INSERT INTO job_status (job_id, owner_id, job_state, project_name, net_error_code, net_error_msg, build_started_at, build_finished_at)
+                                           VALUES (job_id, owner_id, job_state, project_name, net_error_code, net_error_msg, build_started_at, build_finished_at)
+                                           RETURNING *;
+                                       RETURN;
+                                   END
+                               $$ LANGUAGE plpgsql VOLATILE
+                               "#)?;
+
+        migrator.migrate("scheduler",
+                        r#"CREATE OR REPLACE FUNCTION get_job_status_v1 (integer) RETURNS SETOF job_status AS $$
+                               BEGIN
+                                   RETURN QUERY SELECT * FROM job_status
+                                        ORDER BY created_at ASC
+                                        LIMIT $1;
+                                   RETURN;
+                               END
+                               $$ LANGUAGE plpgsql VOLATILE"#)?;
+
+        migrator.migrate(
+            "scheduler",
+            r#"CREATE OR REPLACE FUNCTION delete_job_status_v1 (integer) RETURNS void AS $$
+                              BEGIN
+                                  DELETE FROM job_status
+                                  WHERE id IN (
+                                      SELECT id
+                                      FROM job_status
+                                      ORDER BY created_at ASC
+                                      LIMIT $1
+                                  );
+                              END
+                              $$ LANGUAGE plpgsql VOLATILE"#,
```

Comment:
Yeah these are super simple functions that can go either way - we should look at a high level at our PL/pgSQL vs. SQL usage. PL/pgSQL seems to offer much more flexibility, including the ability to trap errors, better performance with plan caching when things don't get inlined, etc.  It would be nice to just pick one language and stick with it so we can copy/paste more easily and use the patterns for that language.  I've leaned toward pgSQL since that's what the vast bulk of our code is atm.